src/searcher.rs

```diff
@@ -233,6 +235,43 @@ impl Searcher {
         })
     }
 
+    #[pyo3(signature = (search_query, agg_query))]
+    fn aggregate(
+        &self,
+        py: Python,
+        search_query: &Query,
+        agg_query: String,
```

Comment:
I'm looking at the test, and it looks like the `agg_query` parameter is also a dict. I am wondering whether this parameter should be a PyDict instead.  From the python code side, it would be more natural to pass in a dict here rather than have to first convert it into a string.

Reply:
I see it's a little tricky to convert the PyDict into `Aggregations`. I spent a few minutes exploring some options here. I understand why you went with string input, to make use of the deserialization implementations. Is there a way we can receive a dict here, and then just internally proceed via a serialize+deserialize step?  I think this function really should take a dict parameter instead of a string, but I don't want to hold up this PR waiting for wrapper types for the aggregation machinery.

Reply:
Here's something I came up with that passes the test. It uses Python's `json` module because I couldn't figure out how to convert the `PyDict` into a `HashMap` without getting into the `Aggregation` type support:

```rust
    #[pyo3(signature = (search_query, agg_query))]
    fn aggregate(
        &self,
        py: Python,
        search_query: &Query,
        agg_query: &Bound<PyDict>,
    ) -> PyResult<Py<PyDict>> {
        let json = py.import_bound("json")?;
        let serialized = json.call_method1("dumps", (agg_query,))?;
        let serialized = serialized.extract::<&str>()?;

        let agg_str = py.allow_threads(move || {
            let agg_collector = AggregationCollector::from_aggs(
                serde_json::from_str(&serialized).map_err(to_pyerr)?,
                Default::default(),
            );
            let agg_res = self
                .inner
                .search(search_query.get(), &agg_collector)
                .map_err(to_pyerr)?;

            serde_json::to_string(&agg_res).map_err(to_pyerr)
        })?;

        let locals = [("json_str", agg_str)].into_py_dict_bound(py);
        py.run_bound(
            r#"
import json
agg_dict = json.loads(json_str)
"#,
            None,
            Some(&locals),
        )
        .map_err(to_pyerr)?;

        let agg_dict_any = locals.get_item("agg_dict")?.unwrap();
        let agg_dict = agg_dict_any.downcast::<PyDict>()?;

        Ok(agg_dict.clone().unbind())
    }

```

The idea is that we could move forward with this now, and avoid the work of wrapping the underlying types, while still keeping the query argument a `PyDict`, which is what I think the correct API should be.

Reply:
Later we can have a TODO item to use wrapper types for aggregations directly.

Reply:
Thanks for this suggestion. I very much agree and have had a commit ready to go for a couple days but im blocked out of pushes to my branch for a bit :(. Let me resolve this soon so we can get this in!

Reply:
On another note, the call_method you used is much neater than my usage of run_bound so I guess we can refactor that part to use the call API as well!

Reply:
This should be good to go now